src/impls.rs

```diff
@@ -0,0 +1,207 @@
+//! Mutex primitives.
+//!
+//! This module provides impls of the [`ScopedRawMutex`] trait
+//!
+//! [`ScopedRawMutex`]: crate::ScopedRawMutex
+#![allow(clippy::new_without_default)]
+#![allow(clippy::declare_interior_mutable_const)]
+
+use core::marker::PhantomData;
+use core::sync::atomic::{AtomicBool, Ordering};
+
+use crate::{ConstInit, ScopedRawMutex};
+
+#[cfg(feature = "impl-critical-section")]
+pub mod cs {
+
+    use super::*;
+
+    /// A mutex that allows borrowing data across executors and interrupts.
+    ///
+    /// # Safety
+    ///
+    /// This mutex is safe to share between different executors and interrupts.
+    pub struct CriticalSectionRawMutex {
+        taken: AtomicBool,
```

Comment:
kinda wonder if we should be using portable-atomic atomics for these?

Reply:
This would be a non-breaking change, so it's fine to address it later.

Reply:
Not as-is, as we don't use any CAS methods. Agreed this is non-breaking either way.